File: ConstDest.cpp
#include<iostream>
#include<string>
using namespace std;
class Student
{
    private:
        string S_fullName;
        int S_rollNum;
        double S_semPercentage;
        string S_collegeName;
        int S_collegeCode;
    public:
        //Constructor
        Student(string fullName,int rollNum,double semPercentage,string collegeName,int collegeCode)
        {
            S_fullName=fullName;
            S_rollNum=rollNum;
            S_semPercentage=semPercentage;
            S_collegeName=collegeName;
            S_collegeCode=collegeCode;

            cout<<"Student object created."<<endl;
        }
        //Destructor
        ~Student()
        {
            cout<<"Student object destroyed."<<endl;
        }
        void display()
        {
            cout<<"STUDENT DETAILS"<<endl;
            cout<<"Student full name : "<<S_fullName<<endl;
            cout<<"Student roll number : "<<S_rollNum<<endl;
            cout<<"Student sem percentage : "<<S_semPercentage<<endl;
            cout<<"Student college name : "<<S_collegeName<<endl;
            cout<<"Student college code : "<<S_collegeCode<<endl;
        }
};
int main()
{
    string fullName;
    int rollNum;
    double semPercentage;
    string collegeName;
    int collegeCode;
    cout<<"Enter student's full name : ";
    getline(cin,fullName);
    cout<<"Enter student's roll number : ";
    cin>>rollNum;
    cout<<"Enter student's sem percentage : ";
    cin>>semPercentage;
    cin.ignore();   //Ignore the newline character left in the input buffer
    cout<<"Enter student's college name : ";
    getline(cin,collegeName);
    cout<<"Enter student's college code : ";
    cin>>collegeCode;
    Student s1(fullName,rollNum,semPercentage,collegeName,collegeCode);
    s1.display();

    return 0;
}

File: ConstDest.java
import java.util.*;
 class Student{
   private
    String fullName;
    int rollNum;
    double semPercentage;
    String collegeName;
    int collegeCode;
    public
     Student(){
        System.out.println("Created");
     }
    Student(String name,int num,double sem,String coll,int code) {
        fullName = name;
        rollNum = num;
        semPercentage = sem;
        collegeName= coll;
        collegeCode = code;
    }
    void displayPrivate(){
        System.out.println("STUDENT NAME: "+ fullName);
        System.out.println("STUDENT ROLL NO: "+ rollNum);
        System.out.println("STUDENT SEM PERCENTAGE: "+ semPercentage);
        System.out.println("COLLEGE NAME: "+ collegeName);
        System.out.println("COLLEGE CODE: " +collegeCode);
    }
    public void finalize(){
        System.out.println("I'm dead");
    }

public static void main(String[] args){
    Student s = new Student("ch.sai sravani ",533,90,"MVGR",33);
    s.displayPrivate();
    s.finalize();
}
}
